src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -784,6 +795,109 @@ trait BeamHelper extends LazyLogging {
+          (
+            beamConfig.beam.agentsim.agents.rideHail.initialization.parking.filePath,
+            beamConfig.beam.agentsim.taz.parkingFilePath
+          )
+        } else if (fileExist(beamConfig.beam.agentsim.taz.parkingFilePath)) {
+          (beamConfig.beam.agentsim.taz.parkingFilePath, "")
+        } else {
+          ("", "")
+        }
+      }
+
+      if (chargingDepotsFilePath.nonEmpty) {
+        val rand = new Random(beamScenario.beamConfig.matsim.modules.global.randomSeed)
+        val parkingStallCountScalingFactor = beamServices.beamConfig.beam.agentsim.taz.parkingStallCountScalingFactor
+        val (chargingDepots, _) =
+          ParkingZoneFileUtils.fromFile(chargingDepotsFilePath, rand, parkingStallCountScalingFactor)
+
+        var cntChargingDepots = 0
+        var cntChargingDepotsStalls = 0
+        chargingDepots.foreach(
+          parkingZone =>
+            if (parkingZone.chargingPointType.nonEmpty) {
+              cntChargingDepots += 1
+              cntChargingDepotsStalls += parkingZone.stallsAvailable
+          }
+        )
+
+        var cntPublicFastCharge = 0
+        var cntPublicFastChargeStalls = 0
+        if (publicFastChargerFilePath.nonEmpty) {
+          val rand = new Random(beamScenario.beamConfig.matsim.modules.global.randomSeed)
+          val (publicChargers, _) =
+            ParkingZoneFileUtils.fromFile(publicFastChargerFilePath, rand, parkingStallCountScalingFactor)
+
+          publicChargers.foreach(
+            publicCharger =>
+              if (publicCharger.chargingPointType.nonEmpty) {
+                if (ChargingPointType.getChargingPointCurrent(publicCharger.chargingPointType.get) == DC) {
+                  cntPublicFastCharge += 1
+                  cntPublicFastChargeStalls += publicCharger.stallsAvailable
+                }
+            }
+          )
+        }
+
+        writeMetric("beam-run-charging-depots-cnt", cntChargingDepots)
+
+        //if (beamConfig.beam.agentsim.taz.parkingStallChargerInitMethod == "UNLIMITED") {
+        if (cntPublicFastCharge < 0 || cntPublicFastChargeStalls < 0 || cntChargingDepotsStalls < 0) {
+          writeStrMetric("beam-run-public-fast-charge-cnt", "UNLIMITED")
+          writeStrMetric("beam-run-public-fast-charge-stalls-cnt", "UNLIMITED")
+          writeStrMetric("beam-run-charging-depots-stalls-cnt", "UNLIMITED")
+        } else {
+          writeMetric("beam-run-public-fast-charge-cnt", cntPublicFastCharge)
+          writeMetric("beam-run-public-fast-charge-stalls-cnt", cntPublicFastChargeStalls)
+          writeMetric("beam-run-charging-depots-stalls-cnt", cntChargingDepotsStalls)
+        }
+      } else {
+        logger.error(
+          s"Can't read charging information not from 'beamConfig.beam.agentsim.agents.rideHail.initialization.parking.filePath' nor from 'beamConfig.beam.agentsim.taz.parkingFilePath'. Metrics will get 0 values."
+        )
+      }
+    }
   }
```

Comment:
I'd suggest to extract these logic to separate method/class to keep BeamHelper not overloaded

Reply:
done:

https://github.com/LBNL-UCB-STI/beam/pull/2269/files#diff-c31ee0f5d217d37d76585426c8be7c00R1